eos-updater/poll-common.c

```diff
@@ -1190,6 +1217,13 @@ parse_latest_commit (OstreeRepo           *repo,
   else
     *out_version = g_variant_dup_string (version, NULL);
 
+  if (metadata != NULL)
+    release_notes_uri_template = g_variant_lookup_value (metadata, "eos-updater.release-notes-uri", G_VARIANT_TYPE_STRING);
+  if (release_notes_uri_template == NULL)
+    *out_release_notes_uri_template = NULL;
+  else
+    *out_release_notes_uri_template = g_variant_dup_string (release_notes_uri_template, NULL);
```

Comment:
I was just writing some similar code. Doesn't this do the same thing without the intermediate variant?

```
if (metadata == NULL ||
    !g_variant_lookup (metadata, "eos-updater.release-notes-uri", "s", *out_release_notes_uri_template))
  *out_release_notes_uri_template = NULL;
```

I guess you're following the same pattern as above, though.

Reply:
You’re right. I’ve tweaked the commit and added a second commit which cleans up the code above.